packages/jupyterlab-chat-extension/src/chat-commands/providers/user-mention.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import {
+  IChatCommandProvider,
+  IChatCommandRegistry,
+  ChatCommand,
+  IInputModel,
+  Avatar,
+  IUser
+} from '@jupyter/chat';
+import { JupyterFrontEndPlugin } from '@jupyterlab/application';
+import { User } from '@jupyterlab/services';
+import { LabChatContext } from 'jupyterlab-chat';
+
+export const mentionCommandsPlugin: JupyterFrontEndPlugin<void> = {
+  id: 'jupyterlab-chat-extension:mentionCommandsPlugin',
+  description: 'Plugin which adds user mention commands.',
+  autoStart: true,
+  requires: [IChatCommandRegistry],
+  activate: (app, registry: IChatCommandRegistry) => {
+    registry.addProvider(new MentionCommandProvider());
+  }
+};
+
+class MentionCommandProvider implements IChatCommandProvider {
+  public id: string = 'jupyter-chat:mention-commands';
+
+  // regex used to test the current word
+  private _regex: RegExp = /^@[\w-]*:?/;
+
+  async getChatCommands(inputModel: IInputModel) {
+    this._users.clear();
+    const match = inputModel.currentWord?.match(this._regex)?.[0];
+    if (!match) {
+      return [];
+    }
+
+    const users = (inputModel.chatContext as LabChatContext).users;
```

Comment:
This is an example of unsafe type casting encouraged by generic types. If `chatContext` were a union, this could be made type-safe by replacing this with:

```ts
// inputModel: IChatContext
if (inputModel.type !== 'lab') {
  return [];
}

// now inputModel is of type 'LabChatContext'
```